### ISSUE TITLE
[QUERY] is it necessary for changefeed starttime to be truncated to the nearest hour

### ISSUE THREAD

**Issue by gardenia** [Azure/azure-sdk-for-net#40373]
### Library name and version

Azure.Storage.Blobs.ChangeFeed master

### Query/Question

Hi,

I see this code in the various changedeed SDKs, in this case dotnet:

  ./sdk/storage/Azure.Storage.Blobs.ChangeFeed/src/ChangeFeedFactory.cs

```
            // Round start and end time if we are not using the cursor.                                       
            else                                                                                              
            {                                                                                                 
                startTime = startTime.RoundDownToNearestHour();                                               
                endTime = endTime.RoundUpToNearestHour();                                                     
            }
```
            
This has the effect that if I construct a changefeed instance via the factory (and I am not using a continuation) then in the worst case (e.g. if my startTime is 19:58) then I have to consume and throw away almost 1 hour of events in order to get to the "head" of the changefeed.

Yes I fully realise if I am using a continuation this is not the case but my workflow is often to have to create a new connection without a continuation token.

I have 2 questions:

1. is this truncation to the top of the hour technically necessary?  e.g. could it safely start from the start of the latest 15 minute window (for example)?
2. if this is strictly necessary then is there some way I can avoid having to re-consume all the events from the current hour.  Basically all I want is a continuation representing the current head of the event stream but I don't want to have to consume and discard a bunch of events that happened in the current hour.

Thanks.

**Comment by seanmcc-msft** [comment]
@gardenia, yes, this is how the Change Feed is stored in the backend storage service - https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blob-change-feed?tabs=azure-portal#change-feed-segments
